src/lib/acpitbl.c

```diff
@@ -163,21 +182,24 @@ acpitbl_build_rdsp(void) {
 		0x00, 0x00, 0x00, 0x00
 	};
 
-	rdsp = (void *) (((uintptr_t) tb) + 0);
+	rdsp = (char *)tb + 0;
 	/* copy RDSP template to guest memory */
 	memcpy(rdsp, rdsp_tmpl, 36);
 	/* fixup table */
-	acpitbl_write32(rdsp, 0x10, ((uint32_t) (XHYVE_ACPI_BASE + RSDT_OFFSET)));
-	acpitbl_write64(rdsp, 0x18, ((uint64_t) (XHYVE_ACPI_BASE + XSDT_OFFSET)));
+	acpitbl_write32(rdsp, 0x10, (uint32_t)XHYVE_ACPI_BASE + RSDT_OFFSET);
```

Comment:
This ones a bit subtle, depending on how the constants are defined. If `RSDT_OFFSET` were `0xXXXXUL` (which wouldn't be unreasonable) then I think the `uint32_t + unsigned long` would potentially be promoted to an `unsigned long`, despite the first operand having been cast to `uint32_t`, which wasn't quite as intended.

We could write it as `(unt32_t)(FOO + BAR)` (i.e. just fix the whitespace not the braces) but perhaps we would be better off letting the implicit conversion (due to the argument to `acpitbl_write32` taking a `uint32_t`) DTRT? That might mean changing the constants to be unsigned, but arguably that would be correct in its own right?